enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+This enhancement provides an AWS IAM based registration mechanism so that OCM can support EKS-based hub clusters.
+
+## Motivation
+
+For cluster administrators, it is often preferable to leverage their cloud provider’s managed Kubernetes service (e.g. AKS, EKS, IKS, GKE etc) rather than self-managing the Kubernetes control plane and worker nodes - reducing cluster management overhead and complexity.
+However, in some of these environments (for example EKS), the use of CSRs for client authentication is unsupported.
+
+Adopting OCM should not require its users to change how they deploy and manage their hub Kubernetes cluster.
+
+One option is to run [multicluster-controlplane](https://github.com/open-cluster-management-io/multicluster-controlplane) service inside or outside Kubernetes and use that as a hub. However, multicluster-controlplane comes with following limitations:
+- It is not a full Kubernetes controlplane and hence some OCM addons might not work
+- The user is responsible for maintaining HA etcd cluster, instead of being able to leverage managed etcd that comes with EKS controlplane. 
+
+As such, OCM needs to support an alternative registration mechanism to CSRs that is expected to work in EKS, providing maximum compatibility with managed EKS.
+
+### Goals
+
+- The OCM hub cluster can run on EKS.
+- `clusteradm` tooling provides options to select aws-irsa registration while initializing the hub as well as joining a managed cluster to the hub.
+- Make authentication strategy pluggable, making it easier for other cloud providers to add support for their platform in the future.
+
+### Non-Goals
+
+- Support for running a hub cluster on other cloud providers like Google Cloud Platform GKE and Azure AKS, even though this enhancement will make future support for other cloud providers easier.
+- Support for EKS cluster as hub and any other cloud provider managed kubernetes service as a managed cluster. This enhancement aims at using EKS for both managed as well as hub cluster.
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Hub administrator initializes a hub cluster using AWS IRSA authentication strategy
+
+It must be possible for the cluster administrator to specify they wish to authenticate registration requests using `aws-irsa` authentication strategy in the `clusteradm init` command. The default authentication strategy will be `csr`:
+
+```
+% clusteradm init \
+     --wait \
+     --registration-auth=aws-irsa \
+     --context ${CTX_HUB_CLUSTER}
+```
+
+
+#### Story 2 - EKS Hub administrator must initialize a hub using aws-irsa authentication strategy only, using csr authentication strategy will throw an error
+
+`clusteradm init` command must fail if the EKS hub admin tries to initialize the hub, using `csr` authentication or, by leaving it at its default setting.
```

Comment:
Can the hub be non-EKS but wired to use the AWS IAM OIDC server?

Reply:
I am not sure if that would be possible because in addition to AWS OIDC server, we also need IRSA capability of EKS for this solution to work.

Also thinking when would anybody use this kind of setup with non EKS hub and wired with AWS OIDC server.

Reply:
> Also thinking when would anybody use this kind of setup with non EKS hub and wired with AWS OIDC server.

A hub cluster that allows joining from any cloud provider would be beneficial, rather than foreclosing the possibility early.

Reply:
Sure for now, we will keep the hub to support all types of auth for now. We will add support of EKS hub using aws iam only for now.

However, the proposal can be enhanced later to use aws iam auth on non EKS hub as well using TokenReview API. The non EKS hub can get the sts token validated by sending TokenReview request to another EKS cluster.